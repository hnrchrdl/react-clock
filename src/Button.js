// @flow

import React from 'react';

type Props = {
  onClick: () => void,
  text: string,
  active: boolean
};

function Button({ onClick, text, active }: Props) {
  return (
    <span className={`button${active ? ' active' : ''}`} onClick={onClick}>
      {text}
    </span>
  );
}

Button.defaultProps = {
  active: false
};

export default Button;
