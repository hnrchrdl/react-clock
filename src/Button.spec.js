// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

it('should match snapshot', () => {
  const wrapper: Button = shallow(<Button onClick={() => {}} text="text" />);
  expect(wrapper).toMatchSnapshot();
});
