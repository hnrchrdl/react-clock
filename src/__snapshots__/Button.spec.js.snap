// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    active={false}
    onClick={[Function]}
    text="text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "text",
      "className": "button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "text",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "text",
        "className": "button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "text",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
